src/steamship/agents/basic_chat.py

```diff
@@ -0,0 +1,49 @@
+from steamship import SteamshipError
+from steamship.agents.llms.steamship_llm import SteamshipLLM
+from steamship.agents.schema import Action, Agent, AgentContext, FinishAction
+from steamship.agents.utils import build_chat_history
+from steamship.data.tags.tag_constants import RoleTag
+from steamship.plugin.capabilities import ConversationSupport, SystemPromptSupport
+
+DEFAULT_PROMPT = """You are a helpful AI assistant.
+You chat with users are eager to engage on a variety of topics. You answer their questions, provide help thinking
+through challenges they have, and engage as a trusted assistant."""
+
+
+class BasicChatAgent(Agent):
+    """BasicChatAgent implements a conversational agent with no function calling or tool reasoning.
+
+    This Agent class is useful in a number of situations:
+    1) You are looking for an agent with nothing more than open-ended chat, possibly with personality and backstory.
+    2) You are using an LLM that does not support Function-calling natively
+    3) You are using an LLM that, in practice, performs poorly with ReACT-style prompts
+    In these cases, an agent whose only loop does not include an attempt to reason about tool invocation is ideal.
+    A secondary side effect of eliminating tooling is that the less processing must occur before a response begins.
+    """
+
+    PROMPT = DEFAULT_PROMPT
+
+    def __init__(self, llm: SteamshipLLM, **kwargs):
+        # Throw if the user has provided tools as a way to ensure the user understands the limitation of this agent.
+        if "tools" in kwargs:
+            raise SteamshipError(
+                "BasicChatAgent does not support tools. For tool-based agents, please use a base class such as FunctionsBasedAgent."
+            )
+
+        super().__init__(llm=llm, tools=[], **kwargs)
+        self.capabilities = [
+            SystemPromptSupport(),
+            ConversationSupport(),
+        ]
+        self.llm = llm
+
+    def next_action(self, context: AgentContext) -> Action:
+        # Build the Chat History that we'll provide as input to the action
+        messages = build_chat_history(self.PROMPT, self.message_selector, context)
+
+        # call chat() with a hard-coded absence of tools
+        output_blocks = self.llm.generate(messages=messages, tools=[])
```

Comment:
@GitOnUp question while you are also looking at `gpt-4` plugin:  when there are no functions in the call, can we update our call to OpenAI to turn off function-selection? I think by default we use `auto` as the setting, but with the tags / your capability work, we should be able to decide up-front whether or not we even want functions as a possibility in the answer, right?

Reply:
The way that I'd default to doing that is to only turning them on if you get the capability defining functions, yes, and moreso that you'd _disable_ it in the capability if you didn't want them (though the distinction in that specific capability may be moot since if you don't provide functions to call it doesn't make sense to try calling any)